sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Convert.java

```diff
@@ -126,27 +127,41 @@ private static Schema getBoxedNestedSchema(Schema schema) {
           ConvertHelpers.getConvertedSchemaInformation(
               input.getSchema(), outputTypeDescriptor, registry);
       boolean unbox = converted.unboxedType != null;
-      PCollection<OutputT> output =
-          input.apply(
-              ParDo.of(
-                  new DoFn<InputT, OutputT>() {
-                    @ProcessElement
-                    public void processElement(@Element Row row, OutputReceiver<OutputT> o) {
-                      // Read the row, potentially unboxing if necessary.
-                      Object input = unbox ? row.getValue(0) : row;
-                      // The output has a schema, so we need to convert to the appropriate type.
-                      o.output(converted.outputSchemaCoder.getFromRowFunction().apply((Row) input));
-                    }
-                  }));
+      PCollection<OutputT> output;
       if (converted.outputSchemaCoder != null) {
+        output =
+            input.apply(
+                ParDo.of(
+                    new DoFn<InputT, OutputT>() {
+                      @ProcessElement
+                      public void processElement(@Element Row row, OutputReceiver<OutputT> o) {
+                        // Read the row, potentially unboxing if necessary.
+                        Object input = unbox ? row.getValue(0) : row;
+                        // The output has a schema, so we need to convert to the appropriate type.
+                        o.output(
+                            converted.outputSchemaCoder.getFromRowFunction().apply((Row) input));
+                      }
+                    }));
         output =
             output.setSchema(
                 converted.outputSchemaCoder.getSchema(),
                 converted.outputSchemaCoder.getToRowFunction(),
                 converted.outputSchemaCoder.getFromRowFunction());
       } else {
-        // TODO: Support full unboxing and boxing in Create.
-        throw new RuntimeException("Unboxing is not yet supported in the Create transform");
+        SerializableFunction<?, OutputT> convertPrimitive =
+            ConvertHelpers.getConvertPrimitive(converted.unboxedType, outputTypeDescriptor);
+        output =
+            input.apply(
+                ParDo.of(
+                    new DoFn<InputT, OutputT>() {
+                      @ProcessElement
+                      public void processElement(@Element Row row, OutputReceiver<OutputT> o) {
+                        o.output(convertPrimitive.apply(row.getValue(0)));
+                      }
+                    }));
+
+        output.setTypeDescriptor(outputTypeDescriptor);
```

Comment:
do you not need to set a Coder as well?

Reply:
Before I added this line the test was failing because it couldn't find a coder, but just adding the type descriptor fixed it.

Reply:
But I'm only testing unboxing a `Long`, could relying on coder inference be a problem for other types?